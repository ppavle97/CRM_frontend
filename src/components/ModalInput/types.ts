
export interface ModalInputProps {
  state: any;
  setState: any;
  label: string;
  type: string;
}
