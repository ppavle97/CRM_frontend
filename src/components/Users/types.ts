import { User } from "../../store/features/user/userTypes";

export interface StyledUserItemProps {
  active?: boolean;
}

export interface UsersProps {
  users?: User[]
}
