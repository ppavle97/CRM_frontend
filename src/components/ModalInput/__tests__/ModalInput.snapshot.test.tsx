import { render } from "@testing-library/react";
import { ModalInput } from "..";
import { ModalInputProps } from "../types";

describe("ModalInput component", () => {
  const defaultProps: ModalInputProps = {
    label: "Username",
    state: "",
    setState: jest.fn(),
    type: "text",
  };

  it("matches snapshot with default props", () => {
    const { asFragment } = render(<ModalInput {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with updated state", () => {
    const { asFragment, rerender } = render(<ModalInput {...defaultProps} />);

    rerender(<ModalInput {...defaultProps} state="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
