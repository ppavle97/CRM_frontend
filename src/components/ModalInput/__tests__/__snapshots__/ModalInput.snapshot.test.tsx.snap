// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalInput component matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="css-1p58xje"
  >
    <p>
      Username
    </p>
    <input
      placeholder="Type here..."
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`ModalInput component matches snapshot with updated state 1`] = `
<DocumentFragment>
  <div
    class="css-1p58xje"
  >
    <p>
      Username
    </p>
    <input
      placeholder="Type here..."
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;
